pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -167,7 +193,7 @@ type ReconcileInfinispan struct {
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger := eventlog.ValuedLogger(r, "Request.Namespace", request.Namespace, "Request.Name", request.Name)
```

Comment:
I think we can simplify things by having the `eventlog.go` simply be:

```go
type EventLogger interface {
	Logger() *logr.Logger
	EventRecorder() *record.EventRecorder
}

func LogAndSendEvent(el EventLogger, owner runtime.Object, reason, message string) {
	(*el.Logger()).Info(message)
	if *el.EventRecorder() != nil && owner != nil {
		(*el.EventRecorder()).Event(owner, corev1.EventTypeWarning, reason, message)
	}
}
```

In the controllers, we then always create the log instance explicitly (less indirection), i.e.:

```go
r.log = logf.Log.WithName(ControllerName).WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
```

Reply:
the initial idea was to keep in a common place all the initialization stuff for logs and events, honestly I'm this PR is getting old and confused: better to stay simple and limit changes to only the send event feature